prime-dotnet-webapi/Models/SiteRegistration/Site.cs

```diff
@@ -84,5 +92,33 @@ public IEnumerable<DayOfWeek> DaysOpen(DateTimeOffset? atTime = null)
                 .Select(b => b.Day)
                 .Distinct();
         }
+
+        public SiteStatus AddStatus(SiteStatusType siteStatusType)
+        {
+            var newStatus = SiteStatus.FromType(siteStatusType, Id);
+
+            if (SiteStatuses == null)
+            {
+                SiteStatuses = new List<SiteStatus>();
+            }
+            SiteStatuses.Add(newStatus);
+
+            return newStatus;
+        }
+
+        /// <summary>
+        /// Gets the most recent Status of the Site.
+        /// </summary>
+        [NotMapped]
+        [Computed]
+        public SiteStatusType Status
+        {
+            get => (SiteStatuses.Count > 0 ?
+                SiteStatuses
+                .OrderByDescending(s => s.StatusDate)
+                .ThenByDescending(s => s.Id)
+                .FirstOrDefault().StatusType :
+                SiteStatusType.Active);
+        }
```

Comment:
This version of the Property still has a ternary in it, so the generated SQL is still rather complicated:
![original](https://user-images.githubusercontent.com/39168456/124045158-efcd6e80-d9c3-11eb-9ef8-a7aa10a80617.jpg)

You could remove the ternary by doing a Select() instead:
```C#
public SiteStatusType Status
{
    get => SiteStatuses
        .OrderByDescending(s => s.StatusDate)
        .ThenByDescending(s => s.Id)
        .Select(x => (SiteStatusType?)x.StatusType)
        .FirstOrDefault() ?? SiteStatusType.Active;
}
```
but the SQL looks like this:
![Untitled](https://user-images.githubusercontent.com/39168456/124045374-5d799a80-d9c4-11eb-8b95-ea080b043b82.jpg)
which isn't much better.

I personally think this Property probably shouldn't specify a default value for Status if there are none. Since this is a Property on the data model, it should really be a `SiteStatusType?` so it can return null if there is no current status, which more accurately represents the data rather than the business logic surrounding the data. To be clear, I think this is just something to consider going forward, it's not a change I'm requesting for this PR. 

With this change to the code:
```C#
public SiteStatusType? Status
{
    get => SiteStatuses
        .OrderByDescending(s => s.StatusDate)
        .ThenByDescending(s => s.Id)
        .Select(x => (SiteStatusType?)x.StatusType)
        .FirstOrDefault();
}
```
the generated SQL becomes:
![nullable](https://user-images.githubusercontent.com/39168456/124047001-0ecdff80-d9c8-11eb-8644-8e13d6c2ae7e.jpg)
which is much cleaner.

Reply:
Thanks but I'm seeing even more complicated SQL being generated for the ternary situation ... do I have my breakpoint at the wrong spot?  I'll need to show you on Monday @james-hollinger